x-pack/plugins/lens/public/app_plugin/app.test.tsx

```diff
@@ -20,65 +18,70 @@ import {
 import { buildExistsFilter, FilterStateStore } from '@kbn/es-query';
 import type { FieldSpec } from '@kbn/data-plugin/common';
-import { TopNavMenuData } from '@kbn/navigation-plugin/public';
-import { SavedObjectReference } from '@kbn/core/types';
 import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
 import { serverlessMock } from '@kbn/serverless/public/mocks';
 import moment from 'moment';
 import { setState, LensAppState } from '../state_management';
 import { coreMock } from '@kbn/core/public/mocks';
 import { LensSerializedState } from '..';
-import { cloneDeep } from 'lodash';
 import { createMockedField, createMockedIndexPattern } from '../datasources/form_based/mocks';
-import { I18nProvider } from '@kbn/i18n-react';
 import faker from 'faker';
-jest.mock('../editor_frame_service/editor_frame/expression_helpers');
-jest.mock('@kbn/core/public');
+import { screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { VisualizeEditorContext } from '../types';
+import { pluginServices as presentationUtilPluginServices } from '@kbn/presentation-util-plugin/public/services';
+
+presentationUtilPluginServices.getHooks = jest.fn().mockReturnValue({
+  capabilities: {
+    useService: () => ({
+      canAccessDashboards: () => true,
+      canCreateNewDashboards: () => true,
+      canSaveVisualizations: () => true,
+      canSetAdvancedSettings: () => true,
+    }),
+  },
+});
+
 jest.mock('../persistence/saved_objects_utils/check_for_duplicate_title', () => ({
   checkForDuplicateTitle: jest.fn(),
 }));
+jest.mock('lodash', () => ({
+  ...jest.requireActual('lodash'),
+  debounce: (fn: unknown) => fn,
+}));
 
-jest.mock('lodash', () => {
-  const original = jest.requireActual('lodash');
+const defaultSavedObjectId: string = faker.random.uuid();
 
-  return {
-    ...original,
-    debounce: (fn: unknown) => fn,
-  };
-});
+const waitToLoad = async () =>
+  await act(async () => new Promise((resolve) => setTimeout(resolve, 0)));
 
-const sessionIdSubject = new Subject<string>();
+const getLensDocumentMock = (someProps?: Partial<LensDocument>) => ({
+  ...defaultDoc,
+  ...someProps,
+});
 
 describe('Lens App', () => {
-  const defaultSavedObjectId: string = faker.random.uuid();
-
-  function createMockFrame(): jest.Mocked<EditorFrameInstance> {
-    return {
-      EditorFrameContainer: jest.fn((props: EditorFrameProps) => <div />),
-      datasourceMap,
-      visualizationMap,
-    };
-  }
-
-  const navMenuItems = {
-    expectedSaveButton: { emphasize: true, testId: 'lnsApp_saveButton' },
-    expectedSaveAsButton: { emphasize: false, testId: 'lnsApp_saveButton' },
-    expectedSaveAndReturnButton: { emphasize: true, testId: 'lnsApp_saveAndReturnButton' },
-  };
-
-  function makeDefaultProps(): jest.Mocked<LensAppProps> {
-    return {
-      editorFrame: createMockFrame(),
+  let props: jest.Mocked<LensAppProps>;
```

Comment:
One question here.
What's the benefit of having a shared variable and reset it on every test with a `beforeEach` vs generating all the mocks explicitly every time?
It is just code readability (not calling it every time) or there's more?

Reply:
There's way less code to read (like calling makeDefaultProps function and then modifying the properties we want and then pass it to the `renderApp`) with this pattern so I think it's just nicer :)